pkg/notifications/teams.go

```diff
@@ -0,0 +1,50 @@
+package notifications
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func init() {
+	initers = append(initers, func(cfg map[string]string) Notifier {
+		if url, ok := cfg["teams_url"]; ok {
+			notifier := &teamsNotifier{
+				URL: url,
+			}
+			return notifier
+		}
+		return nil
+	})
+}
+
+// teamsNotifier sends notifications to teams or mattermost
+type teamsNotifier struct {
+	URL string
+}
+
+func (s *teamsNotifier) Notify(domain, provider, msg string, err error, preview bool) {
+	var payload struct {
+		Username string `json:"username"`
+		Text     string `json:"text"`
+	}
+	payload.Username = "DnsControl"
+
+	// Format changes as 'preformated' text
+	msg = strings.ReplaceAll(msg, "\n", "\n    ")
+
+	if preview {
+		payload.Text = fmt.Sprintf("**DnsControl Preview %s**\n%s", domain, msg)
+	} else if err != nil {
+		payload.Text = fmt.Sprintf("**DnsControl Error Making Changes %s**\n%s\nError: %s", domain, msg, err)
+	} else {
+		payload.Text = fmt.Sprintf("**DnsControl Successfully Changed %s**\n%s", domain, msg)
+	}
+
+	json, _ := json.Marshal(payload)
```

Comment:
Please check the error and return it upstream.

Reply:
If I return it upstream I'm going to have to change the interface definition as it doesn't currently allow for this. I guess this is going to break anyone else's custom notify types, are you okay with that? 

If so, `Notify(domain, provider string, message string, err error, preview bool) (err error)`, preferences on naming for the two err's?

Reply:
Oops... I misread the signature.  (I should get more sleep :-) )

I'll file a bug to improve this in the future.